src/magic_folder/test/test_config.py

```diff
@@ -547,20 +548,51 @@ def setup_example(self):
 
     @given(
         remote_snapshots(),
+        path_states(),
     )
-    def test_remotesnapshot_roundtrips(self, snapshot):
+    def test_remotesnapshot_roundtrips(self, snapshot, path_state):
         """
         The capability for a ``RemoteSnapshot`` added with
         ``MagicFolderConfig.store_remotesnapshot`` can be read back with
         ``MagicFolderConfig.get_remotesnapshot``.
         """
-        self.db.store_remotesnapshot(snapshot.name, snapshot)
-        loaded = self.db.get_remotesnapshot(snapshot.name)
+        self.db.store_remotesnapshot(snapshot.name, snapshot, path_state)
+        capability = self.db.get_remotesnapshot(snapshot.name)
+        db_path_state = self.db.get_currentsnapshot_pathstate(snapshot.name)
```

Comment:
Seems a bit unfortunate that the symmetry between `store_remotesnapshot` and `get_remotesnapshot` has been broken here and now it takes two Python calls and two SQL statements to get back out what was put in with just one of each.  Maybe this is worth considering a follow-up refactoring to deal with.

Reply:
You may be right. I've linked this comment to https://github.com/LeastAuthority/magic-folder/issues/454#issuecomment-870886295, mentioning that we may want to get these values back atomically.

I didn't implement it here, since there were no non-test consumers that cared about having both, but we probably will want to when change downloading to take this new information into account.